File: backend/app/utils/roles.py
"""Staff role helpers for owner / employee access control."""

STAFF_ROLES = frozenset({"owner", "employee", "admin"})  # admin = legacy alias
OWNER_ROLES = frozenset({"owner", "admin"})  # admin treated as owner during transition


def normalize_role(role: str | None) -> str:
    return (role or "").strip().lower()


def is_staff(role: str | None) -> bool:
    return normalize_role(role) in STAFF_ROLES


def is_owner(role: str | None) -> bool:
    return normalize_role(role) in OWNER_ROLES


def is_employee(role: str | None) -> bool:
    return normalize_role(role) == "employee"

File: frontend/src/lib/roles.ts
/** Staff / customer role helpers (matches backend roles). */

export type UserRole = "customer" | "owner" | "employee" | "admin" | string;

const STAFF_ROLES = new Set(["owner", "employee", "admin"]);
const OWNER_ROLES = new Set(["owner", "admin"]);

export function isStaff(role: string | null | undefined): boolean {
  return STAFF_ROLES.has((role || "").trim().toLowerCase());
}

export function isOwner(role: string | null | undefined): boolean {
  return OWNER_ROLES.has((role || "").trim().toLowerCase());
}

export function isEmployee(role: string | null | undefined): boolean {
  return (role || "").trim().toLowerCase() === "employee";
}

export function roleLabel(role: string | null | undefined): string {
  switch ((role || "").trim().toLowerCase()) {
    case "owner":
    case "admin":
      return "Owner";
    case "employee":
      return "Employee";
    case "customer":
      return "Customer";
    default:
      return role || "Unknown";
  }
}
